streams/readable-byte-streams/general.js

```diff
@@ -326,28 +332,50 @@ promise_test(() => {
   });
 
   const reader = stream.getReader();
-  const readPromise = reader.read();
-  const ignoredReadPromise = reader.read();
+  const p0 = reader.read();
+  const p1 = reader.read();
 
   assert_equals(pullCount, 0, 'No pull() as start() just finished and is not yet reflected to the state of the stream');
 
   return Promise.resolve().then(() => {
     assert_equals(pullCount, 1, 'pull() must have been invoked once');
-    assert_not_equals(byobRequest, undefined, 'byobRequest must not be undefined');
-    assert_true(viewDefined, 'byobRequest.view must not be undefined');
-    assert_equals(viewInfo.constructor, Uint8Array, 'view.constructor should be Uint8Array');
-    assert_equals(viewInfo.bufferByteLength, 16, 'view.buffer.byteLength should be 16');
-    assert_equals(viewInfo.byteOffset, 0, 'view.byteOffset should be 0');
-    assert_equals(viewInfo.byteLength, 16, 'view.byteLength should be 16');
-    return readPromise;
+    const byobRequest = byobRequests[0];
+    assert_true(byobRequest.defined, 'first byobRequest must not be undefined');
+    assert_true(byobRequest.viewDefined, 'first byobRequest.view must not be undefined');
+    const viewInfo = byobRequest.viewInfo;
+    assert_equals(viewInfo.constructor, Uint8Array, 'first view.constructor should be Uint8Array');
+    assert_equals(viewInfo.bufferByteLength, 16, 'first view.buffer.byteLength should be 16');
+    assert_equals(viewInfo.byteOffset, 0, 'first view.byteOffset should be 0');
+    assert_equals(viewInfo.byteLength, 16, 'first view.byteLength should be 16');
+
+    return p0;
   }).then(result => {
+    assert_equals(pullCount, 2, 'pull() must have been invoked twice');
     assert_not_equals(result.value, undefined);
```

Comment:
If you have the energy, it would be nice to have descriptions for these asserts. But don't worry about it, because it was already like this.

Reply:
Done.

---

I'm not sure if I like the assert message to restate the entire check, as in:
```js
assert_equals(viewInfo.byteOffset, 0, 'first view.byteOffset should be 0');
```
If the assert fails, you already get the expected value in the error message anyway:
```
assert_equals: first view.byteOffset should be 0 expected 0 but got 42
```
Perhaps in these cases, we should just describe the value being tested without repeating the expected value? I kind of like the terseness used in some other tests:
```js
assert_equals(view.byteOffset, 0, 'view.byteOffset');
```

Reply:
I make a habit of putting the word 'should' in my descriptions as a way of forcing myself to be clear about what I'm asserting. But certainly in cases such as this it doesn't help readability. The important thing is to have a distinct message, so that someone looking at a failure doesn't have to guess which assert failed.